File: Java/Mahasiswa.java
// Konstruktor.
public class Mahasiswa extends SivitasAkademik
{
    // Private atribut:
    private String nim;
    private String fakultas;
    private String prodi;

    public Mahasiswa()
    {
        this.nim = "";
        this.fakultas = "";
        this.prodi = "";
    }
    
    public Mahasiswa(String nim, String fakultas, String prodi)
    {
        this.nim = nim;
        this.fakultas = fakultas;
        this.prodi = prodi;
    }
    
    // Getter dan Setter.
    // Setter.
    public void set_nim(String nim)
    {
        this.nim = nim;
    }
    public void set_fakultas(String fakultas)
    {
        this.fakultas = fakultas;
    }
    public void set_prodi(String prodi)
    {
        this.prodi = prodi;
    }

    // Getter.
    public String get_nim()
    {
        return this.nim;
    }
    public String get_fakultas()
    {
        return this.fakultas;
    }
    public String get_prodi()
    {
        return this.prodi;
    }
}

File: CPP/Mahasiswa.cpp
// Import library dan files.
#include "SivitasAkademik.cpp"

// Standard namespace.
using namespace std;

// Inisialisasi class.
class Mahasiswa : public SivitasAkademik
{
    // Atribut Private.
    private:
        string nim;
        string prodi;
        string fakultas;

    // Atribut Public.
    public:
        // Konstruktor.
        Mahasiswa()
        {
            // Set default nilai atribut.
            this->nim = "";
            this->prodi = "";
            this->fakultas = "";
        }

        Mahasiswa(string nim, string prodi, string fakultas)
        {
            // Set nilai atribut.
            this->nim = nim;
            this->prodi = prodi;
            this->fakultas = fakultas;
        }

        // Getter dan Setter.
        // Setter.
        void set_nim(string nim)
        {
            this->nim = nim;
        }
        void set_prodi(string prodi)
        {
            this->prodi = prodi;
        }
        void set_fakultas(string fakultas)
        {
            this->fakultas = fakultas;
        }

        // Getter.
        string get_nim()
        {
            return this->nim;
        }
        string get_fakultas()
        {
            return this->fakultas;
        }
        string get_prodi()
        {
            return this->prodi;
        }

        // Method Remove Attribute.
        void hapusDataMahasiswa()
        {
            this->nim = "";
            this->prodi = "";
            this->fakultas = "";
        }

        // Destruktor
        ~Mahasiswa()
        {
            
        }
};